### PR TITLE
Clean-up Gradle, and setup workflows

### PR DESCRIPTION
**Pull request by nomisRev**
This PRs cleans up the Gradle config, and starts re-using the Arrow Gradle Config. This way we can easily setup the publishing.

I've also setup Spotless for formatting the project, and setup some workflows. Converting all dependencies into version catalog.

This should close #1 & #2.

### CODE REVIEW

**Review by ILIYANGERMANOV — APPROVED** [review]
Awesome! Thank you for the setup @nomisRev 🙌

File: src/commonMain/kotlin/arrow.exact/Exact.kt [hunk 1]
```diff
@@ -2,27 +2,21 @@ package arrow.exact
 
 import arrow.core.Either
 
+public interface Exact<A, out R> : ExactEither<ExactError, A, R>
```
Comment (ILIYANGERMANOV) [inline src/commonMain/kotlin/arrow.exact/Exact.kt:5]: @nomisRev Q: Why do we have to right thess explicit `public`?

**Review by ustits — APPROVED** [review]
🚀

File: src/commonMain/kotlin/arrow.exact/Exact.kt (hunk 1 above)
Comment (nomisRev) [reply src/commonMain/kotlin/arrow.exact/Exact.kt:5]: We setup _explicit API_ in Arrow, https://github.com/arrow-kt/arrow-gradle-config/blob/e9298c1f7a0efb1e72194381aedd1a366094cad1/arrow-gradle-config-kotlin/src/main/kotlin/io.arrow-kt.arrow-gradle-config-kotlin.gradle.kts#L21

This is a bit saner for library development.

Comment (ILIYANGERMANOV) [reply src/commonMain/kotlin/arrow.exact/Exact.kt:5]: Makes sense. Thank you!

**Pull request merged**